### PR TITLE
tests: add external_clients scenario (bp #5014)

### PR DESCRIPTION
**Pull request by guits**
This is an automated backport of pull request #5014 done by Mergify.io

### CODE REVIEW

**Comment by guits** [comment]
jenkins test centos-container-external_clients

**Comment by guits** [comment]
jenkins test centos-non_container-external_clients

**Comment by guits** [comment]
jenkins test centos-non_container-external_clients

**Comment by guits** [comment]
jenkins test centos-container-external_clients

**Comment by guits** [comment]
jenkins test centos-container-external_clients

**Comment by guits** [comment]
jenkins test centos-non_container-external_clients

**Comment by guits** [comment]
jenkins test centos-non_container-external_clients

**Comment by guits** [comment]
jenkins test centos-container-external_clients

File: tests/functional/external_clients/container/group_vars/all [hunk 1]
```diff
@@ -0,0 +1,41 @@
+---
+# this is only here to let the CI tests know
+# that this scenario is using docker
+docker: True
+
+containerized_deployment: True
+monitor_interface: "{{ 'eth1' if ansible_distribution == 'CentOS' else 'ens6' }}"
+radosgw_interface: "{{ 'eth1' if ansible_distribution == 'CentOS' else 'ens6' }}"
+radosgw_num_instances: 2
```
Comment (dsavineau) [inline tests/functional/external_clients/container/group_vars/all]: there's no radosgw multi instances in this branch

File: tests/functional/external_clients/container/group_vars/all [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ceph_mon_docker_subnet: "{{ public_network }}"
+ceph_docker_on_openstack: False
+public_network: "192.168.31.0/24"
+cluster_network: "192.168.32.0/24"
+rgw_override_bucket_index_max_shards: 16
+rgw_bucket_default_quota_max_objects: 1638400
+openstack_config: True
+dashboard_enabled: false
```
Comment (dsavineau) [inline tests/functional/external_clients/container/group_vars/all]: dashboard is nautilus+

File: tests/functional/external_clients/container/group_vars/all [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ceph_conf_overrides:
+  global:
+    osd_pool_default_pg_num: 8
+    mon_warn_on_pool_no_redundancy: false
```
Comment (dsavineau) [inline tests/functional/external_clients/container/group_vars/all]: this is an octopus option

File: tests/functional/external_clients/container/vagrant_variables.yml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+# DEPLOY CONTAINERIZED DAEMONS
+docker: True
+
+# DEFINE THE NUMBER OF VMS TO RUN
+mon_vms: 3
+osd_vms: 0
+mds_vms: 0
+rgw_vms: 0
+nfs_vms: 0
+grafana_server_vms: 0
```
Comment (dsavineau) [inline tests/functional/external_clients/container/vagrant_variables.yml]: grafana vms doesn't exist in this branch

File: tox-external_clients.ini [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  ANSIBLE_CONFIG = {toxinidir}/ansible.cfg
+  ANSIBLE_CALLBACK_WHITELIST = profile_tasks
+  ANSIBLE_CACHE_PLUGIN = memory
+  ANSIBLE_GATHERING = implicit
+  # only available for ansible >= 2.5
+  ANSIBLE_STDOUT_CALLBACK = yaml
+#  non_container: DEV_SETUP = True
+  # Set the vagrant box image to use
+  centos-non_container: CEPH_ANSIBLE_VAGRANT_BOX = centos/7
+  centos-container: CEPH_ANSIBLE_VAGRANT_BOX = centos/7
```
Comment (dsavineau) [inline tox-external_clients.ini]: should be centos/atomic-host ?

File: tests/functional/external_clients/group_vars/all [hunk 6]
```diff
@@ -0,0 +1,38 @@
+---
+containerized_deployment: False
+ceph_origin: repository
+ceph_repository: community
+monitor_interface: "{{ 'eth1' if ansible_distribution == 'CentOS' else 'ens6' }}"
+radosgw_interface: "{{ 'eth1' if ansible_distribution == 'CentOS' else 'ens6' }}"
+ceph_mon_docker_subnet: "{{ public_network }}"
+ceph_docker_on_openstack: False
+openstack_config: True
+dashboard_enabled: False
```
Comment (dsavineau) [inline tests/functional/external_clients/group_vars/all]: same comments than for the container setup

**Comment by guits** [comment]
jenkins test centos-non_container-external_clients

**Comment by guits** [comment]
jenkins test centos-container-external_clients

**Comment by guits** [comment]
jenkins test centos-container-external_clients

**Comment by guits** [comment]
jenkins test centos-non_container-external_clients

**Pull request merged**